testkit/src/mill/testkit/UnitTester.scala

```diff
@@ -27,17 +36,36 @@ object UnitTester {
       offline: Boolean = false
   ) = new UnitTester(
     module = module,
-    sourceRoot = sourceRoot,
+    sourceRoot = (Option(sourceRoot), resetSourcePath) match {
+      case (Some(sourceRoot), true) => SourceRoot.ResetAndCopyFrom(sourceRoot)
+      case (Some(sourceRoot), false) => throw new IllegalArgumentException(
+          s"Cannot provide sourceRoot=$sourceRoot when resetSourcePath=false"
+        )
+      case (None, true) => SourceRoot.Reset
+      case (None, false) => SourceRoot.DoNothing
+    },
     failFast = failFast,
     threads = threads,
     outStream = outStream,
     errStream = errStream,
     inStream = inStream,
     debugEnabled = debugEnabled,
     env = env,
-    resetSourcePath = resetSourcePath,
     offline = offline
   )
+
+  /** What to do with [[mill.testkit.TestRootModule.moduleDir]]. */
+  enum SourceRoot {
+
+    /** Do nothing. */
+    case DoNothing
+
+    /** Reset the module dir to an empty directory. */
+    case Reset
+
+    /** Reset the module dir to an empty directory and copy the source tree from the given path. */
+    case ResetAndCopyFrom(path: os.Path)
```

Comment:
Do we really need this? It seems to me like we can just stick with `Reset` and do whatever copying we need to in the body of the integration test

Reply:
I think let's just leave the original 4 cases in place for now, and keep this PR to solving the `bspBuildTargetCompileMerged` failure